File: scripts/algorithms/U/Unique Morse Code Words/Unique Morse Code Words.cpp
// Runtime: 3 ms (Top 85.58%) | Memory: 8.5 MB (Top 62.02%)
class Solution {
public:

  string convert(string st)
  {
     string s1[]={".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
    string s="";
    for(char a:st)
    {
      s+=s1[a - 'a'];

    }
    return s;

  }
    int uniqueMorseRepresentations(vector<string>& words) {

     set<string>st;
      for(int i=0;i<words.size();i++)
      {
      st.insert(convert(words[i]));

      }

      return st.size();
    }
};

File: scripts/algorithms/U/Unique Morse Code Words/Unique Morse Code Words.py
// Runtime: 43 ms (Top 50.2%) | Memory: 17.50 MB (Top 5.12%)

class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        
        # create a dictionary for morse code (You can just copy & paste it! ^.^)
        ENG_to_MORSE = {  
            'a':".-", 'b':"-...", 'c':"-.-.", 'd':"-..", 'e':".",
            'f':"..-.", 'g':"--.", 'h':"....", 'i':"..", 'j':".---",
            'k':"-.-", 'l':".-..", 'm':"--", 'n':"-.", 'o':"---",
            'p':".--.", 'q':"--.-", 'r':".-.", 's':"...", 't':"-",
            'u':"..-", 'v':"...-", 'w':".--", 'x':"-..-", 'y':"-.--", 'z':"--..",
        }
        
        cnt = {}    # dictionary for different transformations
        
        for word in words:      # loop through every word
            
            tmp = ""
            
            for c in word:      # loop through every character
                tmp += ENG_to_MORSE[c]    # convert the word to morse code
                
            if tmp not in cnt:
                cnt[tmp] = 0
            else:
                cnt[tmp] += 1

        return len(cnt)     # return how many different elements in cnt
